Inventory.cs

```diff
@@ -3,34 +3,42 @@
 using System.Linq;
 using System.Text;
 using System.Threading.Tasks;
+using System.Xml.Linq;
 
 namespace SimpleInv
 {
     public class Inventory : IInventory
     {
+        private List<Product> _products;
+
+        public Inventory() {
+            _products = new List<Product>();
+        }
         public void AddProduct(Product product)
         {
-            throw new NotImplementedException();
+            _products.Add(product);
         }
 
         public void RemoveProduct(Product product)
         {
-            throw new NotImplementedException();
+            _products.Remove(product);
         }
 
         public Product Retrieve(string productName)
         {
-            throw new NotImplementedException();
+            return _products.Find(x => x.Name.Equals(productName, StringComparison.OrdinalIgnoreCase));
```

Comment:
what if the product is not found will you depend on find throw exception? or do you think it's better to catch it and throw a custom exception?